docs/gcp-openshift.md

```diff
@@ -1,102 +1,78 @@
 ```bash
 $ oc get secret gcp-credentials -n kube-system --template='{{$v := index .data "service_account.json"}}{{$v}}' | base64 -d - > decoded-gcloud.json
 ```
 
-4. Export Google credentials
+4. Set Google credentials:
 ```bash
 $ export GOOGLE_CREDENTIALS=decoded-gcloud.json
 ```
 
-5. Activate your account
+5. Fetch the values from the credentials file:
+```bash
+$ CLIENT_EMAIL=$(jq -r .client_email < decoded-gcloud.json)
+$ PROJECT_ID=$(jq -r .project_id < decoded-gcloud.json)
+```
+
+6. Activate your account:
 ```bash
-$ gcloud auth activate-service-account  <client_email as per decoded-gcloud.json> --key-file=decoded-gcloud.json
+$ gcloud auth activate-service-account "${CLIENT_EMAIL}" --key-file=decoded-gcloud.json
 ```
 
-6. Set your project
+7. Set your project:
 ```bash
-$ gcloud config set project <project_id as per decoded-gcloud.json>
+$ gcloud config set project "${PROJECT_ID}"
 ```
 
-7. Get the routes to check the domain
+8. Get the routes to check your cluster's domain (everything after `apps`):
 ```bash
 $ oc get routes --all-namespaces | grep console
 openshift-console          console             console-openshift-console.apps.qe.gcp.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
 openshift-console          downloads           downloads-openshift-console.apps.qe.gcp.devcluster.openshift.com                     downloads           http    edge/Redirect          None
 ```
 
-8. Get your zone which was created by the installer
+9. Get your cluster's zone:
 ```bash
-$ gcloud dns managed-zones list | grep <dns name eg- As per the route the section after apps. i.e misalunk-azure.qe.azure.devcluster.openshift.com>
+$ gcloud dns managed-zones list | grep qe.gcp.devcluster.openshift.com
+qe-cvs4g-private-zone qe.gcp.devcluster.openshift.com
 ```
 
-9. Check the zone doesn't have DNS records other than `NS` and `SOA`
+10. Create [ExternalDNS CR](https://github.com/openshift/external-dns-operator/blob/main/config/samples/gcp/operator_v1alpha1_externaldns_openshift.yaml) as follows:
 ```bash
-$ gcloud dns record-sets list --zone=<the zone name you get from step 9>
-```
-
-10. Create a [ExternalDNS CR](https://github.com/openshift/external-dns-operator/blob/main/config/samples/gcp/operator_v1alpha1_externaldns_openshift.yaml)
-```yaml
 $ cat <<EOF | oc create -f -
 apiVersion: externaldns.olm.openshift.io/v1alpha1
 kind: ExternalDNS
 metadata:
   name: sample-gcp
 spec:
   domains:
-    - filterType: Include
-      matchType: Exact
-      name: test-gcp1.qe.gcp.devcluster.openshift.com
+  - filterType: Include
+    matchType: Exact
+    name: qe.gcp.devcluster.openshift.com
   provider:
     type: GCP
   source:
+    type: OpenShiftRoute
     openshiftRouteOptions:
       routerName: default
-    type: OpenShiftRoute
-    fqdnTemplate:
-      - "{{.Name}}.apps.test-gcp1.qe.gcp.devcluster.openshift.com"
```

Comment:
You will need because of the default hostname annotation policy set.

Reply:
https://github.com/openshift/external-dns-operator/pull/122#discussion_r822071001